packages/dds/tree/src/test/utils.ts

```diff
@@ -426,38 +432,48 @@ export class TestTreeProviderLite {
 				logger: this.logger,
 			});
-			const tree = this.factory.create(
-				runtime,
-				TestTreeProviderLite.treeId,
-			) as SharedTreeWithConnectionStateSetter;
+			const tree = this.factory.create(runtime, `tree-${i}`) as ISharedTree;
 			const containerRuntime = this.runtimeFactory.createContainerRuntime(runtime);
-			this.containerRuntimes.add(containerRuntime);
+			this.containerRuntimeMap.set(tree.id, containerRuntime);
 			tree.connect({
 				deltaConnection: runtime.createDeltaConnection(),
 				objectStorage: new MockStorage(),
 			});
-			(tree as Mutable<SharedTreeWithConnectionStateSetter>).setConnected = (
-				connectionState: boolean,
-			) => {
-				containerRuntime.connected = connectionState;
-			};
-			t.push(tree);
+			(tree as Mutable<SharedTreeWithContainerRuntime>).containerRuntime = containerRuntime;
+			t.push(tree as SharedTreeWithContainerRuntime);
 		}
 		this.trees = t;
 	}
 
-	public processMessages(count?: number): void {
-		// In TurnBased mode, flush the messages from all the runtimes before processing the messages.
-		// Note that this does not preserve the order in which the messages were sent. To do so, tests should
-		// flush messages from individual runtimes in the order they were created.
-		if (this.flushMode === FlushMode.TurnBased) {
-			this.containerRuntimes.forEach((containerRuntime) => {
+	/**
+	 * Process the messages across all trees.
+	 * By default, this method also flushes the messages from all trees before processing them. In TurnBased mode,
+	 * messages are not automatically flushed so this ensures that all messages are processed. Flushing can be
+	 * disabled by setting the flush parameter to false. Then it is the responsibility of the test to flush the
+	 * messages before calling this method.
+	 * @param flush - Whether or not to flush the messages before processing them. Defaults to true.
+	 * @remarks
+	 * - For trees whose runtime is paused and / or not connected, these messages will not be processed but queued in
+	 * the runtime. Any queued messages will be processed when the runtime is unpaused and connected.
+	 * - Flushing does not preserve the order in which the messages were sent. To do so, tests should
+	 * call flushMessages on individual trees in the order messages were sent.
+	 */
+	public processMessages(flush: boolean = true): void {
+		if (flush) {
+			this.containerRuntimeMap.forEach((containerRuntime) => {
 				containerRuntime.flush();
 			});
 		}
-		this.runtimeFactory.processSomeMessages(
-			count ?? this.runtimeFactory.outstandingMessageCount,
-		);
+		this.runtimeFactory.processAllMessages();
+	}
+
+	/**
+	 * Process the given count of messages across all trees.
+	 * @remarks trees whose runtime is paused or not connected will queue these messages and not process them. They
+	 * will process the messages when their runtime is unpaused or connected respectively.
+	 */
+	public processSomeMessages(count: number): void {
```

Comment:
It seems a bit confusing to have `processMessages` able to control flushing, but `processSomeMessages` not be able to.
It seems like maybe the best API would be a single method that takes an optional option bag of type `{ readonly count?: number; readonly flush?: boolean }`. What do you think?

Reply:
Renamed these functions to `synchronizeMessages` and paramaterized as per your suggestion.